R/plot_forecast.R

```diff
@@ -0,0 +1,145 @@
+                          truth_as_of = NULL){
+  # validate model
+  if (!missing(model)){
+    if (!model %in% forecast_data$model) {
+      stop("Error in plot_forecast: model is not in forecast_data.")
+    }
+  } else {
+    stop("Error in plot_forecast: Please provide a model parameter.")
+  }
+  
+  # validate target_variable
+  target_variable <- match.arg(target_variable, 
+                               choices = c("Cumulative Cases",
+                                           "Cumulative Deaths",
+                                           "Incident Cases",
+                                           "Incident Deaths"), 
+                               several.ok = FALSE)
+  
+  inc_cum = ifelse(
+    unlist(strsplit(target_variable, " "))[1] == "Cumulative",
+    "cum", 
+    "inc")
+  
+  death_case = ifelse(
+    unlist(strsplit(target_variable, " "))[2] == "Cases",
+    "case",
+    "death")
+
+  if (!(inc_cum %in% forecast_data$inc_cum & death_case %in% forecast_data$death_case)){
+    stop("Error in plot_forecast: target_variable is not in forecast data.")
+  }
+  
+  # validate location fips code
+  all_valid_fips <- covidHubUtils::hub_locations %>%
+    pull(fips)
+  
+  location <- match.arg(location, 
+                         choices = all_valid_fips, 
+                         several.ok = FALSE)
+  
+  if (!location %in% forecast_data$location){
+    stop("Error in plot_forecast: location is not in forecast_data.")
+  }
+  
+  # validate truth_source
+  truth_source <- match.arg(truth_source, 
+                            choices = c("JHU","USAFacts", "NYTimes"), 
+                            several.ok = FALSE)
+  
+  
+  # generate quantiles based on given intervals
+  
+  quantiles_to_plot <- unlist(lapply(intervals, function(interval){
+    c(0.5 - as.numeric(interval)/2,
+     0.5 + as.numeric(interval)/2)
+  }))
+  
+  
+  # prediction interval shades
+  if (!is.null(intervals)){
+    blues <- RColorBrewer::brewer.pal(n=length(quantiles_to_plot)/2+1, "Blues")
+  } else {
+    blues <- RColorBrewer::brewer.pal(n=4, "Blues")
+  }
+  
+  # include truth from remote git hub repo by default
+  # not using truth_as_of if we are loading truth from git hub repos
+  plot_data = covidHubUtils::get_plot_forecast_data (data = forecast_data, 
+                                                     model_to_plot = model,
+                                                     horizons_to_plot = horizon,
+                                                     quantiles_to_plot = quantiles_to_plot,
+                                                     location_to_plot = location,
+                                                     truth_source = truth_source,
+                                                     target_variable = target_variable
+                                                     #truth_as_of = truth_as_of
+                                                     )
+ 
+  
+  graph <- ggplot(data = plot_data)
```

Comment:
Reference packages from which functions come.
```suggestion
  graph <- ggplot2::ggplot(data = plot_data)
```

Also a few other places in this function